src/main/java/network/brightspots/rcv/ResultsWriter.java

```diff
@@ -310,16 +316,25 @@ private void generateSummarySpreadsheet(
     }
     csvPrinter.println();
 
-    // actions don't make sense in individual by-slice results
-    if (isNullOrBlank(sliceId)) {
-      addActionRows(csvPrinter);
-    }
+    final boolean isSlice = !isNullOrBlank(sliceId);
+    csvPrinter.print(isSlice ? "Eliminated*" : "Eliminated");
+    printActionSummary(csvPrinter, roundToEliminatedCandidates);
+
+    csvPrinter.print(isSlice ? "Elected*" : "Elected");
+    printActionSummary(csvPrinter, roundToWinningCandidates);
 
-    // Get all candidates sorted by their first round tally. This determines the display order.
-    List<String> sortedCandidates = roundTallies.get(1).getSortedCandidatesByTally();
+    // Check that all candidates are included in the candidate order
+    Set<String> expectedCandidates = roundTallies.get(1).getCandidates();
+    Set<String> providedCandidates = new HashSet<>(candidateOrder);
+    if (!expectedCandidates.equals(providedCandidates)) {
+      throw new IllegalArgumentException(
+              "Candidate order must include all candidates in the contest. "
+                      + "\nExpected: " + expectedCandidates
+                      + "\nProvided: " + providedCandidates);
+    }
```

Comment:
Should this stanza come first in the method? My concern is that at this point in the method, we've already created a file and started writing to it, so we'd be in kind of an awkward state if this throws.

Reply:
agreed!